core/src/poh_recorder.rs

```diff
@@ -425,4 +433,60 @@ mod tests {
         assert!(poh_recorder.working_bank.is_none());
         assert_eq!(poh_recorder.tick_cache.len(), 3);
     }
+
+    #[test]
+    fn test_reset_current() {
+        let (mut poh_recorder, _entry_receiver) = PohRecorder::new(0, Hash::default());
+        poh_recorder.tick();
+        poh_recorder.tick();
+        assert_eq!(poh_recorder.tick_cache.len(), 2);
+        poh_recorder.reset(poh_recorder.poh.tick_height, poh_recorder.poh.hash);
+        assert_eq!(poh_recorder.tick_cache.len(), 2);
+    }
+
+    #[test]
+    fn test_reset_with_cached() {
+        let (mut poh_recorder, _entry_receiver) = PohRecorder::new(0, Hash::default());
+        poh_recorder.tick();
+        poh_recorder.tick();
+        assert_eq!(poh_recorder.tick_cache.len(), 2);
+        poh_recorder.reset(
+            poh_recorder.tick_cache[0].1,
+            poh_recorder.tick_cache[0].0.hash,
+        );
+        assert_eq!(poh_recorder.tick_cache.len(), 2);
+        poh_recorder.reset(
+            poh_recorder.tick_cache[1].1,
+            poh_recorder.tick_cache[1].0.hash,
+        );
+        assert_eq!(poh_recorder.tick_cache.len(), 2);
```

Comment:
how's this still 2?  I guess I don't understand what reset() does.

Reply:
That value exists in the cache.  So there is nothing to reset.